### ISSUE TITLE
Inconsistency in TF inference for grouped convolutions on CPU

### ISSUE THREAD

**Issue by fg-mindee** [mindee/doctr#454]
## 🐛 Troubles with grouped convolutions on CPU

It seems that TF has troubles with fused grouped convolutions on CPU using `keras.Model.predict` while using the `call` method with `training=False` does run smoothly.
This issue is only happening on CPU, on GPU the script runs smoothly.

## To Reproduce

```python
import os

os.environ['CUDA_VISIBLE_DEVICES'] = '-1'
import tensorflow as tf
from doctr.models import mobilenet_v3_large

samples = tf.zeros((1, 512, 512, 3), dtype=tf.float32)
model = mobilenet_v3_large(input_shape=(512, 512, 3))

# This works
out = model.call(samples, training=False)
# And this throws an error
out = model.predict(samples)
```
which yields
```
---------------------------------------------------------------------------
UnimplementedError                        Traceback (most recent call last)
<ipython-input-1-d329ff911697> in <module>
     11 out = model.call(samples, training=False)
     12 # And this throws an error
---> 13 out = model.predict(samples)

~/miniconda3/lib/python3.8/site-packages/tensorflow/python/keras/engine/training.py in predict(self, x, batch_size, verbose, steps, callbacks, max_queue_size, workers, use_multiprocessing)
   1725           for step in data_handler.steps():
   1726             callbacks.on_predict_batch_begin(step)
-> 1727             tmp_batch_outputs = self.predict_function(iterator)
   1728             if data_handler.should_sync:
   1729               context.async_wait()

~/miniconda3/lib/python3.8/site-packages/tensorflow/python/eager/def_function.py in __call__(self, *args, **kwds)
    887 
    888       with OptionalXlaContext(self._jit_compile):
--> 889         result = self._call(*args, **kwds)
    890 
    891       new_tracing_count = self.experimental_get_tracing_count()

~/miniconda3/lib/python3.8/site-packages/tensorflow/python/eager/def_function.py in _call(self, *args, **kwds)
    954               *args, **kwds)
    955       # If we did not create any variables the trace we have is good enough.
--> 956       return self._concrete_stateful_fn._call_flat(
    957           filtered_flat_args, self._concrete_stateful_fn.captured_inputs)  # pylint: disable=protected-access
    958 

~/miniconda3/lib/python3.8/site-packages/tensorflow/python/eager/function.py in _call_flat(self, args, captured_inputs, cancellation_manager)
   1958         and executing_eagerly):
   1959       # No tape is watching; skip to running the function.
-> 1960       return self._build_call_outputs(self._inference_function.call(
   1961           ctx, args, cancellation_manager=cancellation_manager))
   1962     forward_backward = self._select_forward_and_backward_functions(

~/miniconda3/lib/python3.8/site-packages/tensorflow/python/eager/function.py in call(self, ctx, args, cancellation_manager)
    589       with _InterpolateFunctionError(self):
    590         if cancellation_manager is None:
--> 591           outputs = execute.execute(
    592               str(self.signature.name),
    593               num_outputs=self._num_outputs,

~/miniconda3/lib/python3.8/site-packages/tensorflow/python/eager/execute.py in quick_execute(op_name, num_outputs, inputs, attrs, ctx, name)
     57   try:
     58     ctx.ensure_initialized()
---> 59     tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,
     60                                         inputs, attrs, num_outputs)
     61   except core._NotOkStatusException as e:

UnimplementedError:  Fused conv implementation does not support grouped convolutions for now.
	 [[node mobile_net_v3/inverted_0/sequential/activation_1/Relu (defined at /home/fg/Documents/doctr/doctr/models/backbones/mobilenet/tensorflow.py:140) ]] [Op:__inference_predict_function_6941]

Function call stack:
predict_function

```

## Expected behavior

The script should run smoothly

## Environment

 - DocTR Version: 0.3.1
 - TensorFlow Version: 2.5.0
 - PyTorch & torchvision versions: 1.9.0 (torchvision 0.10.0)
 - OpenCV Version: 4.5.1
 - OS: Ubuntu 20.04.3 LTS
 - How you installed DocTR: source
 - Python version: 3.8
 - CUDA/cuDNN version: CUDA 11.4.100 (cuDNN 8.2.0)
 - GPU models and configuration: NVIDIA GeForce RTX 2070 with Max-Q Design (driver 470.57.02)

**Comment by charlesmindee** [comment]
I seems that they are not likely to fix this in the near future...

**Comment by fg-mindee** [comment]
Yes unfortunately, but we'll have to keep this one open for reference then :/

**Comment by felixdittrich92** [comment]
@charlesmindee @fg-mindee same problem if we try to export to SavedModel format (only mobilenet arch)

**Comment by frgfm** [comment]
So the upstream problem about grouped conv on CPU is now resolved for TF 2.9 👍 
We're gonna have to look at updating our version specifiers
